src/urbit/app/chess.hoon

```diff
@@ -265,8 +279,24 @@
             ==
           %=  this
             games  %+  ~(put by games)  game-id.action
-                   [game position |2.u.game-state]
+                   [game position fen-repetition threefold-draw-available |4.u.game-state]
           ==
+        %claim-threefold
+          =/  game  (~(get by games) game-id.action)
+          ?~  game
+            :_  this
+            =/  err
+              "no active game with id {<game-id.action>}"
+            :~  [%give %poke-ack `~[leaf+err]]
+            ==
+          :-  :~  :*  %give  %fact  ~[/game/(scot %da game-id.action)/updates]
+                      %chess-update
+                      !>([%claim-threefold game-id.action])
+                  ==
+              ==    
+          %=  this
+            games  (~(put by games) game-id.action u.game(claim-threefold-draw !claim-threefold-draw.u.game))
+          == 
```

Comment:
Claiming the threefold repetition draw doesn't need to be its own `chess-action`, it's just a `chess-move` like any other. Only one thing complicates it: the opponent may need to make the move and claim the draw at the same time, which `chess-move` doesn't support right now. It needs to be extended to have the option to claim a draw as part of a regular piece move.

Generic code for handling end moves and the 50-move draw already exists in `with-position` in `../lib/chess.hoon` (called by `try-move` in `../app/chess.hoon', itself called by both `on-poke` for our moves and `on-agent` for opponent moves).

However, this will not do for the threefold repetition draw, as `with-position` looks only at the current position in a vacuum. If we could extend the `chess-position` type to track move repetitions in a field like `ply-50-move-rule`, that would solve the issue. Unfortunately, that's not possible: tracking the 50-move rule simply entails incrementing the count when a move without a pawn or capture takes place, and resetting the count to 0 when a pawn move or capture occurs. Unlike this simple arithmetic, counting position repetitions requires comparing against past positions.

Therefore, the next best place is probably in `try-move`, which already handles some other end states (e.g. checkmate and stalemate). I see that you've already modified the `try-move` code to take the `map` of prior FEN positions. We may as well simplify the call:
```
++  try-move
  |=  [game=active-game-state move=chess-move]
  ^-  [new=(unit active-game-state) cards=(list card)]
  ...
```

Reply:
Note based on our conversation here: https://github.com/ashelkovnykov/urbit-chess/pull/40#discussion_r975547508

With a slightly different name and a slightly different action, this poke could still be useful for setting an automatic preference on the backend regarding whether to automatically claim threefold repetition/50-move draws or not.